drivers/flash/flash_stm32h7x.c

```diff
@@ -126,20 +158,66 @@ int flash_stm32_wait_flash_idle(const struct device *dev)
 	return 0;
 }
 
-static int erase_sector(const struct device *dev, uint32_t sector)
+static struct flash_stm32_sector get_sector(off_t offset)
+{
+	struct flash_stm32_sector sector;
+
+#ifdef DUAL_BANK
+	bool bank_swap;
+	/* Check whether bank1/2 are swapped */
+	bank_swap = (READ_BIT(FLASH->OPTCR, FLASH_OPTCR_SWAP_BANK)
+			== FLASH_OPTCR_SWAP_BANK);
+
+	if ((offset < (REAL_FLASH_SIZE / 2)) && !bank_swap) {
+		sector.sector_index = offset / FLASH_SECTOR_SIZE;
+		sector.bank = 1;
+	} else if ((offset >= BANK2_OFFSET) && bank_swap) {
+		sector.sector_index = (offset - BANK2_OFFSET)
+					 / FLASH_SECTOR_SIZE;
+		sector.bank = 1;
+	} else if ((offset < (REAL_FLASH_SIZE / 2)) && bank_swap) {
+		sector.sector_index = offset / FLASH_SECTOR_SIZE;
+		sector.bank = 2;
+	} else if ((offset >= BANK2_OFFSET) && !bank_swap) {
+		sector.sector_index = (offset - BANK2_OFFSET)
+					/ FLASH_SECTOR_SIZE;
+		sector.bank = 2;
+	} else {
+		sector.sector_index = 0;
+		sector.bank = 0;
+	}
+#else
+	if (offset < REAL_FLASH_SIZE) {
+		sector.sector_index = offset / FLASH_SECTOR_SIZE;
+		sector.bank = 1;
+	} else {
+		sector.sector_index = 0;
+		sector.bank = 0;
+	}
+#endif
+
+	return sector;
+}
+
+static int erase_sector(const struct device *dev, int offset)
 {
 	FLASH_TypeDef *regs = FLASH_STM32_REGS(dev);
 	int rc;
 	uint32_t tmp;
 	volatile uint32_t *cr;
+	struct flash_stm32_sector sector = get_sector(offset);
 
-	if (sector < FLASH_SECTOR_TOTAL) {
+	switch (sector.bank) {
+	case 1:
 		cr = &regs->CR1;
+		break;
 #ifdef DUAL_BANK
```

Comment:
Maybe put it in `get_sector()` function? Leads to better code readability.

Reply:
Done